nomt/src/seglog/segment_filename.rs

```diff
@@ -0,0 +1,120 @@
+use anyhow::{Context, Result};
+
+pub fn format(prefix: &str, segment_id: u32) -> String {
+    // The format string specifies a 10-digit number, so we pad with leading zeros from
+    // the left. This assumes that segment_id is a 32-bit integer, which is confirmed by
```

Comment:
can you humor me with a back-of-envelope calculation about whether 32 bits is enough?

Reply:
Assume you create a segment for each commit. Personally, I am assuming arguably a conservative case where we do 3 commits per second. If so, it would take around 10 years (2^32 s / 3) of such usage to exhaust the whole range. That, together with the another assumption that we assume frequent resyncs, it looks fine.

Moreover, to actually trigger a problem, you need start overwriting the data, which can only happen without pruning, but with any reasonable segment size you would exhaust modern disks first before you get to that point. 

There is still a slight problem though: we should use wrapping_add for segment_id addition, and in fact, in the original rollback PR it did use wrapping_add, but it was lost after this rewrite. It should be fine without it in practice because without debug_assertions on it would still wrap around but ideally we still file a fix.

Reply:
Actually, now thinking about it, it would be a tad more complicated than using wrapping_add. This is because we rely on IDs to imply ordering upon recovery. So if a recovery does happen when there are segment ID values around the both sides of u32::MAX, then the segments would be presented in a wrong order and the recovery would fail because it would detect the mismatch in record IDs.

I would still say that 32-bit is ought to be enough for everyone and this would be fine for now.

Reply:
Yeah, this is convincing enough, but I wonder - why not use a u64. is there a downside to it?